python/ambassador/envoy/v2/v2ready.py

```diff
@@ -0,0 +1,153 @@
+    def access_log(cls, config: 'V2Config') -> List[dict]:
+        access_log: List[dict] = []
+
+        # Use sane access log spec in JSON
+        if config.ir.ambassador_module.envoy_log_type.lower() == "json":
+            log_format = config.ir.ambassador_module.get('envoy_log_format', None)
+            if log_format is None:
+                log_format = {
+                    'start_time': '%START_TIME%',
+                    'method': '%REQ(:METHOD)%',
+                    'path': '%REQ(X-ENVOY-ORIGINAL-PATH?:PATH)%',
+                    'protocol': '%PROTOCOL%',
+                    'response_code': '%RESPONSE_CODE%',
+                    'response_flags': '%RESPONSE_FLAGS%',
+                    'bytes_received': '%BYTES_RECEIVED%',
+                    'bytes_sent': '%BYTES_SENT%',
+                    'duration': '%DURATION%',
+                    'upstream_service_time': '%RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)%',
+                    'x_forwarded_for': '%REQ(X-FORWARDED-FOR)%',
+                    'user_agent': '%REQ(USER-AGENT)%',
+                    'request_id': '%REQ(X-REQUEST-ID)%',
+                    'authority': '%REQ(:AUTHORITY)%',
+                    'upstream_host': '%UPSTREAM_HOST%',
+                    'upstream_cluster': '%UPSTREAM_CLUSTER%',
+                    'upstream_local_address': '%UPSTREAM_LOCAL_ADDRESS%',
+                    'downstream_local_address': '%DOWNSTREAM_LOCAL_ADDRESS%',
+                    'downstream_remote_address': '%DOWNSTREAM_REMOTE_ADDRESS%',
+                    'requested_server_name': '%REQUESTED_SERVER_NAME%',
+                    'istio_policy_status': '%DYNAMIC_METADATA(istio.mixer:status)%',
+                    'upstream_transport_failure_reason': '%UPSTREAM_TRANSPORT_FAILURE_REASON%'
+                }
+
+                tracing_config = config.ir.tracing
+                if tracing_config and tracing_config.driver == 'envoy.tracers.datadog':
+                    log_format['dd.trace_id'] = '%REQ(X-DATADOG-TRACE-ID)%'
+                    log_format['dd.span_id'] = '%REQ(X-DATADOG-PARENT-ID)%'
+
+            access_log.append({
+                'name': 'envoy.access_loggers.file',
+                'typed_config': {
+                    '@type': 'type.googleapis.com/envoy.config.accesslog.v2.FileAccessLog',
+                    'path': config.ir.ambassador_module.envoy_log_path,
+                    'json_format': log_format
+                }
+            })
+        else:
+            # Use a sane access log spec
+            log_format = config.ir.ambassador_module.get('envoy_log_format', None)
+
+            if not log_format:
+                log_format = 'ACCESS [%START_TIME%] \"%REQ(:METHOD)% %REQ(X-ENVOY-ORIGINAL-PATH?:PATH)% %PROTOCOL%\" %RESPONSE_CODE% %RESPONSE_FLAGS% %BYTES_RECEIVED% %BYTES_SENT% %DURATION% %RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)% \"%REQ(X-FORWARDED-FOR)%\" \"%REQ(USER-AGENT)%\" \"%REQ(X-REQUEST-ID)%\" \"%REQ(:AUTHORITY)%\" \"%UPSTREAM_HOST%\"'
```

Comment:
This is a lot of data to duplicate from `v3listener.py`.

Reply:
Yes. This is also duplicated from other places.
I've discussed with @kflynn and I agree this should be extracted and the access log format stuff should be part of a library reused by other parts.
I didn't want to tackle this on this specific review because I wanted to focus on improving the /ready endpoint, also because refactoring the log format would impact other modules, and I didn't want to start mixing things, and finally as v2 is going to go away at some point, having duplicates between v2 and v3 does not sound like a big deal as v2 is just going to be removed eventually.